getting_started/scripting/gdscript/gdscript_basics.rst

```diff
@@ -1463,49 +1463,88 @@ freed.
 Signals
 ~~~~~~~
 
-Signals are a way to send notification messages from an object that
-other objects can listen to in a generic way. Create custom signals for
-a class using the ``signal`` keyword.
+Signals are a tool to emit messages from an object that other objects can react
+to. To create custom signals for a class, use the ``signal`` keyword.
 
 ::
 
-   # Signal with no arguments
-   signal your_signal_name
+   # A signal named health_depleted
+   signal health_depleted
 
-   # Signal with two arguments
-   signal your_signal_name_with_args(a, b)
+   # A signal that forwards two arguments
+   signal health_changed(old_value, new_value)
 
-These signals may be connected to methods in the same manner as you connect
-built-in signals of nodes such as :ref:`class_Button` or :ref:`class_RigidBody`.
+.. note:: 
 
-Here's an example that creates a custom signal in one script and connects
-the custom signal to a method in a separate script, using the
+   Signals are a `Callback
+          <https://www.wikiwand.com/en/Callback_(computer_programming)>`
+          mechanism. They also fill the role of Observers, a common programming
+          pattern. For more information, read the `Observer
+          tutorial<http://gameprogrammingpatterns.com/observer.html>`_ in the
+          Game Programming Patterns ebook.
```

Comment:
Why is the whole paragraph indented?
Also, I think it would be better to link to Wikipedia instead of Wikiwand.